src/main/java/com/datastax/cdm/cql/statement/TargetUpsertRunDetailsStatement.java

```diff
@@ -111,11 +112,13 @@ public Collection<PartitionRange> getPendingPartitions(long prevRunId, JobType j
             }
         }
 
-        final Collection<PartitionRange> pendingParts = new ArrayList<PartitionRange>();
+        final List<PartitionRange> pendingParts = new ArrayList<PartitionRange>();
         pendingParts.addAll(getPartitionsByStatus(prevRunId, TrackRun.RUN_STATUS.NOT_STARTED.toString(), jobType));
         pendingParts.addAll(getPartitionsByStatus(prevRunId, TrackRun.RUN_STATUS.STARTED.toString(), jobType));
         pendingParts.addAll(getPartitionsByStatus(prevRunId, TrackRun.RUN_STATUS.FAIL.toString(), jobType));
         pendingParts.addAll(getPartitionsByStatus(prevRunId, TrackRun.RUN_STATUS.DIFF.toString(), jobType));
```

Comment:
Should we use this opportunity to refactor these lines as below,
```suggestion
        // Use an array of statuses for iteration
        String[] statuses = {
            TrackRun.RUN_STATUS.NOT_STARTED.toString(),
            TrackRun.RUN_STATUS.STARTED.toString(),
            TrackRun.RUN_STATUS.FAIL.toString(),
            TrackRun.RUN_STATUS.DIFF.toString()
        };
        for (String status : statuses) {
            pendingParts.addAll(getPartitionsByStatus(prevRunId, status, jobType));
        }
```